pkg/endpoint/bpf.go

```diff
@@ -1051,32 +1118,22 @@ func (e *Endpoint) applyPolicyMapChanges(regenContext *regenerationContext, hasN
 		return nil
 	}
 
-	// Add policy map entries before deleting to avoid transient drops
+	// Add policy map entries before deleting to avoid transient drops. If there
+	// isn't enough space to add all the entries before deleting some, then delete
+	// first.
```

Comment:
Question: should we log something (maybe at a warning level) when we invert the insertion/deletion order? Or at least clarify in the docs that when we are close to the limit (but not yet in the condition to lock the endpoint) some transient drops might happen due to this inversion?